pkg/mountutil/volumestore/volumestore.go

```diff
@@ -154,6 +156,28 @@ func (vs *volumeStore) Get(name string) (*native.Volume, error) {
 	return &entry, nil
 }
 
+func (vs *volumeStore) GetWithSize(name string) (*native.Volume, error) {
+	vol, err := vs.Get(name)
+	if err != nil {
+		return nil, err
+	}
+	walkDirFn := func(_ string, info os.FileInfo, err error) error {
```

Comment:
NOLGTM here. We may have a lot of tiny directories, and the recursive subdirectories may need lots of time.  So maybe it's a good idea to add a context here, and we can add extra timeout control outside the func 

See https://github.com/moby/moby/blob/master/pkg/directory/directory_unix.go#L14-L55

Reply:
Won't the user be able to cancel the command with ctrl+c?

Reply:
Yes, they can cancel by sending SIGINT. but maybe it's an excellent choice to set a default timeout for this potential long-time action. I guess

Reply:
I would really like to avoid a context combined with --size.
We can have a timeout for this function
```
nerdctl volume inspect --size --timeout 10s <volume-name> and this times out after a second. 
```
If timeout is specified and command takes more than 10s, we can timeout with an error message.

LMK if this sounds good.

Reply:
SGTM！

Reply:
> --timeout

`timeout(1)` might be enough.
https://man7.org/linux/man-pages/man1/timeout.1.html